src/connectedk8s/azext_connectedk8s/_params.py

```diff
@@ -46,7 +50,13 @@ def load_arguments(self, _):
         c.argument('no_proxy', options_list=['--proxy-skip-range'], arg_group='Proxy', help='List of URLs/CIDRs for which proxy should not to be used.')
         c.argument('proxy_cert', options_list=['--proxy-cert'], arg_group='Proxy', type=file_type, completer=FilesCompleter(), help='Path to the certificate file for proxy')
         c.argument('disable_proxy', options_list=['--disable-proxy'], arg_group='Proxy', action='store_true', help='Disables proxy settings for agents')
-        c.argument('auto_upgrade', options_list=['--auto-upgrade'], help='Flag to enable/disable auto upgrade of arc agents. By default, auto upgrade of agents is enabled.', arg_type=get_enum_type(["true", "false"]))
+        c.argument('auto_upgrade', options_list=['--auto-upgrade'], help='Flag to enable/disable auto upgrade of Arc agents. By default, auto upgrade of agents is enabled. This flag must be false when using custom registry repository', arg_type=get_enum_type(["true", "false"]))
+        c.argument('container_registry_repository', options_list=['--registry-repository'], arg_group='Agent upgrade and container registry', help='Custom container registry repository.')
+        c.argument('container_registry_username', options_list=['--registry-username'], arg_group='Agent upgrade and container registry', help='Custom container registry repository username.')
+        c.argument('container_registry_password', options_list=['--registry-password'], arg_group='Agent upgrade and container registry', help='Custom container registry repository password.')
+        c.argument('container_registry_agent_version', options_list=['--agent-version'], arg_group='Agent upgrade and container registry', help='Version of agent to update the helm charts to. Required for custom container registry')
+        c.argument('default_registry', options_list=['--use-default-registry'], arg_group='Agent upgrade and container registry', action='store_true', help='Flag to move from custom container registry repository to default registry i.e., Microsoft Container Registry.')
+        c.argument('container_registry_anonymous_pull', options_list=['--anonymous-pull'], arg_group='Agent upgrade and container registry', action='store_true', help='Flag to move from custom container registry with credentials to custom container registry with anonymous pull.')
```

Comment:
Similar suggestion here: --registry-anonymous-pull instead of --anonymous-pull

Reply:
flag becomes too long hence its --agent-version. Suggested length is 22 char

Reply:
okay